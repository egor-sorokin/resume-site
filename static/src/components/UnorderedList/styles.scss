@import '../../styles/base/_vars.scss';

.unorderedlist {
  &__item {
    text-align: center;

    &:nth-of-type(n+2) {
      margin-top: 8px;
    }
  }
  &__text {
    color: $mercuryLight;
    font-size: 16px;
    font-weight: 300;
  }
}
