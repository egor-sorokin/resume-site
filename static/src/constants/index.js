export const DEFAULT_API_URL = 'http://127.0.0.1:8080/api/v.1.0/';
export const URL_PATH_PERSON_DATA = 'person';
export const URL_PATH_PROJECTS = 'projects';
