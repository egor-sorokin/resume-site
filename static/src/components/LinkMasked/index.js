import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const LinkStretched = ({linkData, cssClasses, target}) => (
  <div className="link-masked">
    <span className={`link-masked__placeholder ${cssClasses}`}>{linkData.text}</span>
    <div className="link-masked__mask">
      <a
        href={linkData.url}
        target={target}
        className={`link-masked__text ${cssClasses}`}
      >
        {linkData.text}
      </a>
    </div>
  </div>
);

LinkStretched.propTypes = {
  linkData: PropTypes.shape({}),
  cssClasses: PropTypes.string,
  target: PropTypes.string
};

LinkStretched.defaultProps = {
  linkData: {},
  cssClasses: '',
  target: ''
};

export default LinkStretched;