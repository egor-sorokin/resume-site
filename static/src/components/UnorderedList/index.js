import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const UnorderedList = ({title, items}) => {
  const unorderedListItems = items.map((item) => (
    <li key={item.id}
        className="unorderedlist__item">
      <a href={item.url}
         target="_blank"
         className="unorderedlist__text">{item.name}</a>
    </li>
  ));

  return (
    <div className="unorderedlist">
      {title.length > 0 && (<h5 className="unorderedlist__title">{title}</h5>)}
      <ul className="unorderedlist__list">
        {unorderedListItems}
      </ul>
    </div>
  )
};

UnorderedList.propTypes = {
  title: PropTypes.string,
  items: PropTypes.instanceOf(Array)
};

UnorderedList.defaultProps = {
  title: '',
  items: []
};

export default UnorderedList;
