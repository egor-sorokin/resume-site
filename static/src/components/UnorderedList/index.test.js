// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import UnorderedList from './index';
import LinkMasked from '../LinkMasked/index';
import {LINK_MASKED} from '../../constants/index';


describe('<UnorderedList />', () => {
  let component;
  const title = 'title';
  let items = [
    {
      "id": 1,
      "text": "test text 1",
      "url": "test.urlone"
    },
    {
      "id": 2,
      "text": "test text 2",
      "url": "test.urltwo"
    }
  ];

  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<UnorderedList />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 1 component', () => {
      expect(component.children()).toHaveLength(1);
    });
  });


  describe('props tests', () => {
    beforeEach(() => {
      component = mount(<UnorderedList items={items}/>);
    });


    it('renders 1 simple link', () => {
      expect('items' in component.props()).toBeTruthy();
      expect(component.find('li').first().childAt(0).is('a')).toBeTruthy();
    });


    it('renders masked link', () => {
      console.log(component.debug());
      component.setProps({type: LINK_MASKED});
      console.log(component.debug());

      expect('items' in component.props()).toBeTruthy();
      expect('type' in component.props()).toBeTruthy();
      expect(component.prop('type')).toEqual(LINK_MASKED);
      expect(component.find(LinkMasked)).toHaveLength(2);
    });


    it('renders a title for list', () => {
      component.setProps({title: title});
      expect('items' in component.props()).toBeTruthy();
      expect('title' in component.props()).toBeTruthy();
      expect(component.prop('title')).toEqual(title);
      expect(component.find('.unorderedlist__title')).toHaveLength(1);
    });
  });
});
