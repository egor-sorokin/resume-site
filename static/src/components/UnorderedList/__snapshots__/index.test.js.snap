// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnorderedList /> base tests matches snapshot 1`] = `
<div
  className="unorderedlist"
>
  <ul
    className="unorderedlist__list"
  />
</div>
`;
